src/Vector3.h

```diff
@@ -121,11 +126,7 @@ class Vector3
 	{
 		// Perform a bitwise comparison of the contents - we want to know whether this object is exactly equal
 		// To perform EPS-based comparison, use the EqualsEps() function
-		return (
-			(memcmp(&x, &a_Rhs.x, sizeof(x)) == 0) &&
-			(memcmp(&y, &a_Rhs.y, sizeof(y)) == 0) &&
-			(memcmp(&z, &a_Rhs.z, sizeof(z)) == 0)
-		);
+		return !((x != a_Rhs.x) || (y != a_Rhs.y) || (z != a_Rhs.z));
```

Comment:
It does, but the behavior it warns about (bitwise/strict comparison of floats) is exactly what is wanted in this case, so the warnings are false positives.
The original version is way unsafer and does the same (bitwise/strict comparison of floats). The only difference is that clang doesn't recognize it as such and thus doesn't trigger a warning.